### DISCUSSION TITLE
Clash with M2Doc

### DISCUSSION THREAD

**Discussion by ASH-SYSTEMS** [eclipse-sirius/sirius-desktop#241] [General]
I am trying to migrate a previous solution to Sirius 7.3. The solution uses M2Doc.
When I try to resolve target definitions for the solution there is a conflict which says:
"Only one of the following can be installed at once: "
Software being installed: M2Doc Sirius Integration 3.3.2.202312041107 (org.obeonetwork.m2doc.sirius.feature.feature.group 3.3.2.202312041107)
Software being installed: Sirius Core Runtime 7.3.0.202311270829 (org.eclipse.sirius.runtime.feature.group 7.3.0.202311270829)

Can you advise?

**Comment by ylussaud** [comment]
I'm maintaining M2Doc and I didn't test M2Doc with Sirius 7.3.0 yet. I need to widen the Sirius dependency range. There will be a release supporting Sirius 7.3.x for the Capella 7.0.0 release. I have a related subject with an other dependency, you can have a look at this [issue](https://github.com/ObeoNetwork/M2Doc/issues/513) to follow evolution on this topic. Then this issue is closes you will be able to use a nightly build, see the following update site:

https://s3-eu-west-1.amazonaws.com/obeo-m2doc-releases/master/nightly/repository

At the moment I don't have any visibility on when this will be done. The only workaround for now is to stick with Sirius 7.2.x.

**Comment by ASH-SYSTEMS** [comment]
Hi,

Thank you for the response.
Following your advice, I tried to use Sirius 7.2.1 and 7.2.0. Both do not
have "org.obeonetwork.m2doc.sirius.feature.group" but instead
"org.obeonetwork.m2doc.sirius.feature.feature.group" so I used that instead.
However, the problem remains:
[image: image.png]

Any advice?

Thanks,
Avi

**Comment by ylussaud** [comment]
I checked the range and you need to use Sirius 7.1.x. I'll widen the ranges for a nightly build.

**Comment by ylussaud** [comment]
I widened the Sirius range to 7.4.0 excluded and added a target platform for Sirius 7.2.1 and ran basic tests with this version. I didn't test with Sirius 7.3.0 yet. You can check the nightly build:

https://s3-eu-west-1.amazonaws.com/obeo-m2doc-releases/master/nightly/repository

**Reply by ASH-SYSTEMS** [reply]
Thank you!
The nightly build does seem to fix the conflicting installation issue. I tested it wth Sirius 7.3.
Can you please provide a target platform definition file for using M2Doc?
Also, specifically, I now have issue with the "import org.obeonetwork.m2doc.sirius.M2DocSiriusUtils;"
The M2DocSiriusUtils package seems to be missing - was it replaced by another name?
[I have lines like "siriusOpt.setName(M2DocSiriusUtils.SIRIUS_SESSION_OPTION)" used in my code]

**Reply by ylussaud** [reply]
Target platforms used to build M2Doc are available [here](https://github.com/ObeoNetwork/M2Doc/tree/master/releng/org.obeonetwork.m2doc.targetplatforms). You need to add the update site for the version of M2Doc you want (probably the nightly version) to the target you selected.

**Reply by ASH-SYSTEMS** [reply]
Thanks.
Everything seems to work.
